spec/models/miq_expression_spec.rb

```diff
@@ -1631,4 +1631,30 @@
       end
     end
   end
+
+  describe "field_in_sql?" do
+    it "should returns false for model.virtualfield" do
+      field = "ManageIQ::Providers::InfraManager::Vm-archived"
+      expression = {"=" => {"field" => field, "value" => "true"}}
+      expect(described_class.new(expression).field_in_sql?(field)).to eq(false)
```

Comment:
Unfortunately – no, valid expression with field we are testing is required. 
Chain of calls:   #field_in_sql(field)  ->  #field_from_virtual_reflection(field)  ->  #col_details(field) 
->  self.class.get_cols_from_expression(@exp, @preprocess_options)
and field parameter lost in last call